### DISCUSSION TITLE
Does Quarto support newer Mermaid shapes?

### DISCUSSION THREAD

**Discussion by roaldarbol** [quarto-dev/quarto-cli#11483] [Q&A] [answered]
### Description

I had a rummage around the Mermaid docs, and found some new shapes that I really liked (https://mermaid.js.org/syntax/flowchart.html#complete-list-of-new-shapes). However, they do not work for me in Quarto. Is there anything I can do as an end-user to upgrade the version of Mermaid, or am I simply doing something silly?

Example:
````qmd
:::{#fig-intro-2}
```{mermaid}
flowchart TB
  A --> B@{ shape: procs, label: "Low-level behaviours"}
```
:::
````

(When `Preview`ing in VS Code, I get `undefined`)

**Comment by cscheid** [comment, marked as answer]
Nothing you can do (aside from forking Quarto, that is). This is the relevant file: https://github.com/quarto-dev/quarto-cli/blob/main/src/resources/formats/html/mermaid/mermaid.js

We update every dependency at most once a release, and ideally once every release.

**Reply by roaldarbol** [reply]
Fair enough - I suspected as much and that makes complete sense. May I be cheeky and ask when you plan to update the dependency/release? 😅

**Reply by roaldarbol** [reply]
[Seems they were introduced in 11.3](https://mermaid.js.org/syntax/flowchart.html#expanded-node-shapes-in-mermaid-flowcharts-v11-3-0).

**Reply by mcanouil** [reply]
For 1.7, very likely.

**Reply by stx-chris** [reply]
I would like to re-open this question, as `Mermaid.js` is now at `11.6.0` and a lot of shapes and functions, especially to Flowchart, have been introduced and fixed since the currently supported version `11.2.0`. Is a chore update planned for Quarto `1.8` (current latest release as of June 2025 is 1.8.13)?

**Reply by cscheid** [reply]
https://github.com/quarto-dev/quarto-cli/pull/12939
